eng/testing/tests.mobile.targets

```diff
@@ -146,6 +153,26 @@
 
   </Target>
 
+  <Target Condition="'$(TargetOS)' == 'Browser' and '$(EnableAggressiveTrimming)' == 'true'"
```

Comment:
Why do you need both TargetOS and EnableAggressiveTrimming conditions

Reply:
We have only tested on wasm, so it's extra protection ;-).

Reply:
EnableAggressiveTrimming is already conditioned, right?

Reply:
Right, the main condition to configure trimming is for `EnableAggressiveTrimming` to be set. If we remove the `'$(TargetOS)' == 'Browser'` check, should we be assuming the flag will only be used for Browser since the changes in this PR were only checked on Browser? Would it be better to broaden the condition after working out publish trimmed on iOS and Android?

Reply:
There should be almost no difference between trimming tests runner for ios and browser

Reply:
Removed the `TargetOS` specific condition, and instead vary `TrimmerRootAssembly` based on `TargetOS`.